sheetstorm/tests/test_unit_generate_test_data.py

```diff
@@ -0,0 +1,331 @@
+        self.assertTrue(GenerateTestDataCommand._set_user_factory_parameters(user_type).get("is_staff"))
+
+    @parameterized.expand([CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name])
+    def test_employee_and_manager_should_not_have_staff_property(self, user_type):
+        self.assertFalse(GenerateTestDataCommand._set_user_factory_parameters(user_type).get("is_staff"))
+
+    @parameterized.expand(
+        [CustomUser.UserType.ADMIN.name, CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name]
+    )
+    def test_any_user_should_not_have_superuser_property_except_superuser(self, user_type):
+        self.assertFalse(GenerateTestDataCommand._set_user_factory_parameters(user_type).get("is_superuser"))
+
+    def test_superuser_should_have_superuser_property(self):
+        self.assertTrue(GenerateTestDataCommand._set_user_factory_parameters(self.superuser).get("is_superuser"))
+
+    @parameterized.expand([CustomUser.UserType.ADMIN.name, superuser])
+    def test_admin_and_superuser_should_have_user_type_coded_as_admin(self, user_type):
+        self.assertEqual(
+            GenerateTestDataCommand._set_user_factory_parameters(user_type).get("user_type"),
+            CustomUser.UserType.ADMIN.name,
+        )
+
+    @parameterized.expand([CustomUser.UserType.EMPLOYEE.name, CustomUser.UserType.MANAGER.name])
+    def test_employee_and_manager_should_have_user_type_coded_as_based_of_user_type_name(self, user_type):
+        self.assertEqual(GenerateTestDataCommand._set_user_factory_parameters(user_type).get("user_type"), user_type)
+
+    @parameterized.expand(
+        [
+            (CustomUser.UserType.ADMIN.name, 2),
+            (CustomUser.UserType.EMPLOYEE.name, 5),
+            (CustomUser.UserType.MANAGER.name, 3),
+        ]
+    )
+    def test_result_of_create_user_function_should_be_specified_number_of_users_in_database(
+        self, user_type, number_of_users_to_create
+    ):
+        GenerateTestDataCommand().create_user(user_type, number_of_users_to_create)
+
+        self.assertEqual(CustomUser.objects.filter(user_type=user_type).count(), number_of_users_to_create)
+
+    @parameterized.expand(
+        [
+            (CustomUser.UserType.ADMIN.name, 3),
+            (CustomUser.UserType.EMPLOYEE.name, 3),
+            (CustomUser.UserType.MANAGER.name, 3),
+        ]
+    )
+    def test_despite_filled_database_result_of_create_user_function_should_be_specified_number_of_users_in_database(
+        self, user_type, number_of_users_to_create
+    ):
+        mock_number_of_users_in_database_less_than_next_request = 2
+
+        self._generate_test_users(mock_number_of_users_in_database_less_than_next_request, user_type)
+
+        GenerateTestDataCommand().create_user(user_type, number_of_users_to_create)
+
+        self.assertEqual(CustomUser.objects.filter(user_type=user_type).count(), number_of_users_to_create)
```

Comment:
What is your approach to line separation here? I'm just curious, because I would keep lines 97 and 99 together, treating them both equally as a sort of pre-test setup. What's the idea behind keeping them separated?